packages/admin/admin/src/dataGrid/columnsManagement/DataGridColumnsManagement.tsx

```diff
@@ -0,0 +1,181 @@
+import { gridColumnVisibilityModelSelector, type GridPinnedColumnPosition, useGridApiContext, useGridSelector } from "@mui/x-data-grid-pro";
+import { type ChangeEvent, useMemo } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { type ThemedComponentBaseProps } from "../../helpers/ThemedComponentBaseProps";
+import { Body, Divider, List, ListHeader, Root } from "./DataGridColumnsManagement.sc";
+import { DataGridColumnsManagementListItem } from "./DataGridColumnsManagementListItem";
+
+export type DataGridColumnsManagementProps = ThemedComponentBaseProps<{
+    root: "div";
+    body: "div";
+    list: typeof MuiList;
+    listHeader: typeof ListSubheader;
+    divider: typeof MuiDivider;
+}>;
+
+export type DataGridColumnsManagementClassKey = "root" | "body" | "list" | "listHeader" | "divider";
+
+export const DataGridColumnsManagement = (inProps: DataGridColumnsManagementProps) => {
+    const { sx, className, slotProps = {} } = useThemeProps({ props: inProps, name: "CometAdminDataGridColumnsManagement" });
+
+    const apiRef = useGridApiContext();
+    const columnVisibilityModel = useGridSelector(apiRef, gridColumnVisibilityModelSelector);
+    const columns = apiRef.current.getAllColumns();
+
+    const pinnedColumns = apiRef.current.getPinnedColumns();
+
+    const toggleColumn = (event: ChangeEvent<HTMLInputElement>) => {
+        const { name: field } = event.target;
+        const column = columns.find((value) => value.field === field);
+        if (column?.hideable) {
+            const isCurrentlyVisible = columnVisibilityModel[field] !== false;
+
+            const newVisibilityModel = {
+                ...columnVisibilityModel,
+                [field]: !isCurrentlyVisible,
+            };
+
+            apiRef.current.setColumnVisibilityModel(newVisibilityModel);
+        }
+    };
+
+    const pinnedLeftColumns = useMemo(() => {
+        return columns.filter((column) => {
+            return pinnedColumns.left?.includes(column.field) ?? false;
+        });
+    }, [columns, pinnedColumns.left]);
+
+    const scrollableColumns = useMemo(() => {
+        return columns.filter((column) => {
+            return !pinnedColumns.left?.includes(column.field) && !pinnedColumns.right?.includes(column.field);
+        });
+    }, [columns, pinnedColumns.left, pinnedColumns.right]);
+
+    const pinnedRightColumns = useMemo(() => {
+        return columns.filter((column) => {
+            return pinnedColumns.right?.includes(column.field) ?? false;
+        });
+    }, [columns, pinnedColumns.right]);
+
+    const handleOnPinColumn: (field: string, position: GridPinnedColumnPosition) => void = (field, position) => {
+        const column = columns.find((value) => value.field === field);
+        if (!column || !column.pinnable) {
+            return;
+        }
+
+        const currentPosition = apiRef.current.isColumnPinned(field);
+
+        if (currentPosition === false || position !== currentPosition) {
```

Comment:
I'd prefer a separate, named variable for this to improve readability, like `isAlreadyPinned`

Reply:
`isAlreadyPinned` is not totally correct, because it checks if the column `isAlreadyPinned` or `isPinnedToAnotherPosition`

Extracted it to a variable and named it `shouldPin`